### PR TITLE
[REFACTOR] Clearly Separate Common MessageFrame Fields

### PR DESCRIPTION
**Pull request by RatanRSur**
I'm looking to add the common field of initial warmed up accounts/storage for EIP-2930 and thought that this would make it cleaner for myself and future additions to the `MessageFrame`

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java [hunk 1]
```diff
@@ -320,9 +320,7 @@ private MessageFrame(
     this.transactionHash = transactionHash;
     this.revertReason = revertReason;
 
-    this.warmedUpAddresses = new HashSet<>();
-    warmedUpAddresses.add(sender);
-    warmedUpAddresses.add(contract);
+    this.warmedUpAddresses = new HashSet<>(List.of(sender, contract));
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java:323]: This winds up creating a new List object for each iteration, and creates a loop that can simply be unrolled to what is originally there.  Yes, it's two more lines of code but a lot less VM work.

Comment (RatanRSur) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java:323]: Sure thing, I'll remove it

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java [hunk 2]
```diff
@@ -295,77 +296,61 @@ public TransactionProcessingResult processTransaction(
...
+                        .contract(to)
+                        .contractAccountVersion(
+                            maybeContract
+                                .map(AccountState::getVersion)
+                                .orElse(Account.DEFAULT_VERSION))
+                        .inputData(transaction.getPayload())
+                        .code(
+                            new Code(maybeContract.map(AccountState::getCode).orElse(Bytes.EMPTY)));
+                  })
+              .orElseGet(
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java:340]: This reduces the readability.  The if block makes it clear that there is one path for contract creation and one for normal transactions.  Without that you need to have deep knowledge of the transactions that a contract creation TX lacks a to field.  I don't mind the common field refactoring but I think the over-use of optional here reduces the accessibility to future maintainers.

Comment (RatanRSur) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetTransactionProcessor.java:340]: Yeah, it's true that the `isContractCreation` method does help with understanding what's going on. I can change the uncommon part back to the if/else.

**Review by shemnon — APPROVED** [review]

**Pull request merged**